### ISSUE TITLE
utility crate: Indexed DB

### ISSUE THREAD

**Issue by fitzgen** [rustwasm/gloo#2]
Should have a utility crate for working with Indexed DB in a Rust-y, idiomatic way.

**Comment by Pauan** [comment]
It should also fully support Futures, since all the IndexedDB APIs are asynchronous.

**Comment by prasannavl** [comment]
I think approaching with a similar high level semantics, like Dexie in js, would probably be the key in making it actually usable. Indexed DB is tedious to use as it is, and I'm afraid of even thinking about a Rust-y future chain.

**Comment by fitzgen** [comment]
> high level semantics

I agree that we want to get there, but I am a big fan of creating layers that go from raw `-sys` style bindings up to callbacks, then to futures, and finally to a high level semantics along the way. This allows folks to reuse whichever layer is appropriate if they are building an alternative style of high level API or if they need direct bindings access, etc.

**Comment by prasannavl** [comment]
Absolutely in agreement. Just wanted to clarify on what layers are being set as expectations in the scope of  `gloo` and what's built outside of it.

**Comment by fitzgen** [comment]
See also the discussion happening in #27 about layers of APIs.

**Comment by devashishdxt** [comment]
I'm experimenting with a high level indexed db wrapper (futures based) [here](https://github.com/devashishdxt/rexie). Feedback is welcome. 😁
